### PR TITLE
#820: return Out Of Stock Status if downloadable has no links

### PR DESCRIPTION
**Pull request by vovsky**
PR fixes bug when stock status of downloadable product is In_Stock despite product has no links.

### Fixed Issues (if relevant)
1. magento/graphql-ce#820: Downloadable Product should be OUT OF STOCK if there are no links just samples saved

### Note
1. I had to add dependency to "magento/module-downloadable", please let me know if you see the way how to avoid it.
2. Deprecated StockStatusInterface is still used for retrieving constant StockStatusInterface::STATUS_IN_STOCK. Please let me know if there is a better replacement for this.

### CODE REVIEW

File: app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php [hunk 1]
```diff
@@ -47,7 +48,20 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
 
         $stockStatus = $this->stockStatusRepository->get($product->getId());
         $productStockStatus = (int)$stockStatus->getStockStatus();
+        $stockStatusByProductType = StockStatusInterface::STATUS_IN_STOCK;
 
-        return $productStockStatus === StockStatusInterface::STATUS_IN_STOCK ? 'IN_STOCK' : 'OUT_OF_STOCK';
```
Comment (lenaorobei) [inline app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php:51]: I think we need to check something like `$product->isInStock() && $product->isSaleable()`.

**Review by lenaorobei — COMMENTED** [review]
Please address my comment and cover this case with api functional test.
Thanks.

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5830](https://jira.corp.magento.com/browse/ENGCOM-5830) has been created to process this Pull Request

File: app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php (hunk 1 above)
Comment (vovsky) [reply app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php:51]: @lenaorobei 
Unfortunately $product->isInStock() and $product->isSaleable() returns true for Out of Stock simple product, both methods check product status and not a stock :(
So currently I don't see how can we avoid using deprecated StockStatus interfaces

Comment (lenaorobei) [reply app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php:51]: Yes, my recommendation was incorrect, sorry about that. Looks like the root cause is somewhere `\Magento\CatalogInventory\Model\StockRegistryProvider::getStockStatus` Need to debug and understand, why it does not take `saleable` into account.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please address found issue.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen to continue working on it. Thank you.

**Pull request closed without merging**